File: Kotlin/comparing.kt
const val YES = 1
const val NO = -1
const val EQ = 0

fun main() {
    val S = input()
    output(S)
}

fun input(): String = readLine() ?: ""

fun output(S: String) {
    val result = check(S)
    print(
        when (result) {
            YES -> "Да"
            NO -> "Нет"
            else -> "Одинаково"
        }
    )
}

fun check(S: String): Int {
    var numOfVowels = 0
    var numOfConsonants = 0
    
    for (s in S) {
        when {
            isVowel(s) -> numOfVowels++
            s != ' ' -> numOfConsonants++
        }
    }
    
    return when {
        numOfVowels > numOfConsonants -> YES
        numOfVowels < numOfConsonants -> NO
        else -> EQ
    }
}

fun isVowel(s: Char): Boolean = s in "aeiouyAEIOUY"

File: Ruby/comparing.rb
YES = 1
NO = -1
EQ = 0

def input(str)
  str.replace(gets.chomp)
end

def output(str)
  result = check(str)
  print result == YES ? "Да" : (result == NO ? "Нет" : "Одинаково")
end

def check(str)
  numOfVowels = 0
  numOfConsonants = 0
  numOfOthers = 0

  str.each_char do |s|
    if isVowel(s)
      numOfVowels += 1
    elsif s != ' '
      numOfConsonants += 1
    else
      numOfOthers += 1
    end
  end

  numOfVowels > numOfConsonants ? YES : (numOfVowels < numOfConsonants ? NO : EQ)
end

def isVowel(s)
  vowels = "aeiouyAEIOUY"
  vowels.include?(s)
end

str = ""
input(str)
output(str)